rocketc/rocketc.py

```diff
@@ -538,3 +596,72 @@ def _get_list_groups(self, user_id, auth_token, query=""):
             groups = groups["groups"]
             for group in groups:
                 yield group["name"]
+
+    def _get_user_messages(self, group_name, latest="", oldest="", count=100):
+        """
+        Gets the messages from a user's private group.
+        """
+        api = self.api_rocket_chat
+        rocket_chat_group = api.search_rocket_chat_group(group_name)
+
+        if not rocket_chat_group['success']:
+            return None
+
+        group_id = rocket_chat_group['group']['_id']
+        messages = api.get_groups_history(room_id=group_id, latest=latest,
+                                          oldest=oldest, count=count)
+
+        if not messages["success"]:
+            return None
+
+        return [message for message in messages["messages"]
+                if message["u"]["username"] == self.user_data["username"]]
+
+    def _filter_by_reaction_and_user_role(self, messages, reaction):
+        """
+        Returns generator with filtered messages by a given reaction
+        """
+        for message in messages:
```

Comment:
what about using the filter function? https://docs.python.org/2/library/functions.html#filter

Reply:
What is the advantage ?

Reply:
For me is more legible, but according to this post, you can use also list comprehension: https://stackoverflow.com/questions/3013449/list-comprehension-vs-lambda-filter/3013686

Reply:
Ok after our discussion, just reduce the the two ifs to just one